python-main.js

```diff
@@ -785,69 +795,75 @@ function web_editor(config) {
                     return;
                 }
 
-
                 $("#flashing-overlay-error").html('<div>' + err + '</div><div>Please restart your micro:bit and try again</div><a href="#" onclick="flashErrorClose()">Close</a>');
             });
         }
     }
 
     function setupHterm(){
-               hterm.defaultStorage = new lib.Storage.Local();
-               var t = new hterm.Terminal("opt_profileName");
-               t.options_.cursorVisible = true;
-
-               var daplinkReceived = false;
-
-               t.onTerminalReady = function() {
-                   var io = t.io.push();
-
-                   io.onVTKeystroke = function(str) {
-                        window.daplink.serialWrite(str);
-                   };
-
-                   io.sendString = function(str) {
-                        window.daplink.serialWrite(str);
-                   };
-
-                   io.onTerminalResize = function(columns, rows) {
-                   };
-
-
-               };
-
-               $("#editor-container").hide();
-               $("#repl").show();
-               $("#request-repl").show();
-
-               t.decorate(document.querySelector('#repl'));
-               t.installKeyboard();
-
-               // Recalculate terminal height
-               $("#repl > iframe").css("position", "relative");
-               $("#repl").attr("class", "hbox flex1");
-
-               window.daplink.on(DAPjs.DAPLink.EVENT_SERIAL_DATA, function(data) {
-                       t.io.print(data); // first byte of data is length
-                       daplinkReceived = true;
-               });
-
-               /* Don't do this automatically
-               // Send ctrl-C to get the terminal up
-               var attempt = 0;
-               var getPrompt = setInterval(
-                       function(){
-                            daplink.serialWrite("\x03");
-                            console.log("Requesting REPL...");
-                            attempt++;
-                            if(attempt == 5 || daplinkReceived) clearInterval(getPrompt);
-                        }, 200);
-               */
+       if (REPL == null) {
+         hterm.defaultStorage = new lib.Storage.Local();
```

Comment:
If we are constantly creating a new local storages instead of retrieving old ones does this mean we are polluting the broswer local storage with old data?
@microbit-sam do you know what is the lifecycle of this if we are not retrieving the local storage? Do we need to use local storage at all?

Reply:
I think we set up the possibility of using local storage, but never used it as we didn't set any preferences

I'm not completely certain how `lib.Storage` handles local storage, but font size doesn't need to be persistent across sessions then we could use `lib.Storage.Memory()` instead

Reply:
https://github.com/macton/libdot/blob/be8c52cb78b27f941534abcff6f111c7ca4235e0/src/lib_storage_local.js#L13

`lib.Storage.Local()` will fetch `window.localStorage` so it will be the same localStorage instance across sessions

Reply:
Right so that's not "allocating a block of memory" that is only used during this session and then left to rot, correct?
If we can actually do without using localstorage (since we are not reusing data between sessions) that is even better, but maybe that should go into a different task.

Reply:
Correct

If we use `hterm.defaultStorage = new lib.Storage.Local();` it sticks them in an object that will only last as long as the page

https://github.com/reterm/libdot/blob/bbd1659e711ee07565673617641ffac448751d50/src/storage/Memory.js#L11

Reply:
Great, thanks for confirming!